### PR TITLE
chore: reinit with auth, tenancy and bloat removal

### PR DESCRIPTION
**Pull request by dirkdev98**
Reinit scaffold removing pretty much everything and adding tenancy and auth support out of the box.

TODO:
- [x] Go through diff and detect unnecessary changes
- [x] Add Playwright setup + CI
  - Requires lightbasenl/scaffold-backend#64
  - Debug webkit error
- [x] Write documentation
  - Scaffold development (requires scaffold-backend if auth or other api things are needed)
  - Scaffold usage ( document the `TODO(platform)` items
  - Scaffold contents
- [x] GitHub action for docker builds
- [ ] Act on feedback

### CODE REVIEW

**Review by d1sc1pl1n3 — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+COPY --from=builder /app/package.json ./package.json
+COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
+COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static
+
+USER nextjs
+EXPOSE 3000
 
-CMD ["yarn", "start"]
+ENV PORT 3000
+CMD ["node", "server.js"]
```
[the remark this answers is not in the document]
Comment (dirkdev98) [reply Dockerfile:51]: This is a feature of standalone Next.js builds, where generates the `server.js`

[the remark this answers is not in the document]
Comment (dirkdev98) [reply Dockerfile:51]: https://nextjs.org/docs/advanced-features/output-file-tracing

File: src/pages/api/_health.api.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  res.status(200).end();
```
[the remark this answers is not in the document]
Comment (dirkdev98) [reply src/pages/api/_health.api.ts]: 78d81bb8a9b85dc42cb473b4ec93679db54e2e31

File: next.config.js [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+/** @type {import("next").NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  swcMinify: true,
+  output: "standalone",
   preset: {
-    transpileModules: ["yup", "framesync", "popmotion", "style-value-types"],
-    ignoreModules: ["next"],
+    // transpileModules: [],
+    // ignoreModules: [],
```
[the remark this answers is not in the document]
Comment (dirkdev98) [reply next.config.js:12]: Didn't now if I needed them, will take a look what the side effects are of an empty array here in the next-preset

[the remark this answers is not in the document]
Comment (dirkdev98) [reply next.config.js]: Build seems to work, did uncomment them.

File: src/middleware.api.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    return NextResponse.redirect(new URL("/412", req.url));
+  }
+
+  // TODO(platform): include all private paths in this if-statement
+  if (url.pathname.startsWith("/private")) {
+    if (!req.cookies.has("accessToken")) {
+      // TODO(platform): redirect to correct login url
+      return NextResponse.redirect(new URL("/", req.url));
+    }
+  }
```
[the remark this answers is not in the document]
Comment (dirkdev98) [reply src/middleware.api.ts:43]: /cc @rodymolenaar , @bjarn

[the remark this answers is not in the document]
Comment (rodymolenaar) [reply src/middleware.api.ts:43]: I agree with @dirkdev98 here. We want to do as little in this middleware as possible, as it's executed on every single request (event `_next` calls). Permissions can be checked through hooks or and even potentially be handled through other hooks' `onError` callback.

**Review by rodymolenaar — APPROVED** [review]

File: src/middleware.api.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dirkdev98) [reply src/middleware.api.ts:43]: FYI merging this and creating a separate issue

**Pull request merged**